pkg/build/util/jenkins.go

```diff
@@ -0,0 +1,164 @@
+package util
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift/origin/pkg/api/latest"
+	"github.com/openshift/origin/pkg/client"
+	serverapi "github.com/openshift/origin/pkg/cmd/server/api"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	kclient "k8s.io/kubernetes/pkg/client/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+// NewJenkinsPipelineTemplate returns a new JenkinsPipelineTemplate.
+func NewJenkinsPipelineTemplate(ns string, conf serverapi.JenkinsPipelineConfig, kubeClient *kclient.Client, osClient *client.Client) *JenkinsPipelineTemplate {
+	return &JenkinsPipelineTemplate{
+		Config:          conf,
+		TargetNamespace: ns,
+		kubeClient:      kubeClient,
+		osClient:        osClient,
+	}
+}
+
+// JenkinsPipelineTemplate stores the configuration of the
+// JenkinsPipelineStrategy template, used to instantiate the Jenkins service in
+// given namespace.
+type JenkinsPipelineTemplate struct {
+	Config          serverapi.JenkinsPipelineConfig
+	TargetNamespace string
+	kubeClient      *kclient.Client
+	osClient        *client.Client
+	items           []resourceMapping
+	ProcessErrors   []error
+	CreateErrors    []error
+}
+
+// Process processes the Jenkins template. If an error occurs
+func (t *JenkinsPipelineTemplate) Process() *JenkinsPipelineTemplate {
```

Comment:
This is not a style we usually use in our code.  I would recommend you return different object, or simply return an error + the objects to create.  Process errors block create errors.

```
objects, err := t.Process()
if err != nil {
}
if err := t.Instantiate(objects); err != nil {
}
```

Instantiate doesn't really even have anything to do with the template, so again, this isn't wrong but it's stylistically incompatible.